.gitpod.Dockerfile

```diff
@@ -1,30 +1,26 @@
-FROM gitpod/workspace-full:2023-01-16-03-31-28
-
-RUN mkdir -p ~/.local/bin
-RUN curl -L https://github.com/stellar/soroban-tools/releases/download/v20.3.0/soroban-cli-20.3.0-x86_64-unknown-linux-gnu.tar.gz | tar xz -C ~/.local/bin soroban
-RUN chmod +x ~/.local/bin/soroban
-RUN curl -L https://github.com/mozilla/sccache/releases/download/v0.3.3/sccache-v0.3.3-x86_64-unknown-linux-musl.tar.gz | tar xz --strip-components 1 -C ~/.local/bin sccache-v0.3.3-x86_64-unknown-linux-musl/sccache
-RUN chmod +x ~/.local/bin/sccache
-
-RUN curl -L https://github.com/watchexec/cargo-watch/releases/download/v8.1.2/cargo-watch-v8.1.2-x86_64-unknown-linux-gnu.tar.xz | tar xJ --strip-components 1 -C ~/.local/bin cargo-watch-v8.1.2-x86_64-unknown-linux-gnu/cargo-watch
-
-ENV RUSTC_WRAPPER=sccache
-ENV SCCACHE_CACHE_SIZE=5G
-ENV SCCACHE_DIR=/workspace/.sccache
+FROM gitpod/workspace-full:latest
 
 # Remove the existing rustup installation before updating due to:
 # https://github.com/gitpod-io/workspace-images/issues/933#issuecomment-1272616892
 RUN rustup self uninstall -y
 RUN rm -rf .rustup
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain none -y
 
-RUN rustup install 1.69
-RUN rustup target add --toolchain 1.69 wasm32-unknown-unknown
-RUN rustup component add --toolchain 1.69 rust-src
-RUN rustup default 1.69
+RUN rustup install stable
+RUN rustup target add --toolchain stable wasm32-unknown-unknown
+RUN rustup component add --toolchain stable rust-src
+RUN rustup default stable
 
 RUN sudo apt-get update && sudo apt-get install -y binaryen
 
 # Enable sparse registry support, which will cause cargo to download only what
 # it needs from crates.io, rather than the entire registry.
 ENV CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
+
+# Install cargo binstall
+RUN curl -L --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash
+RUN cargo binstall -y stellar-cli cargo-watch sccache
```

Comment:
Thanks for reordering the tools to install after the other things, that makes more sense. 👍🏻 

Could we keep the existing curl binaries install process? We shouldn't need to change the process, just update the versions I would expect?

Reply:
Hi @leighmcculloch, I don't oppose using the existing method, but I think using `cargo binstall` would make everything look cleaner. In fact, `cargo binstall` also downloads binary files from the GitHub release page. If we need to specify a version, we can do that here too. However, in this scenario, I believe providing users with the latest version should be sufficient, and this way we won't have to update this file every time.

Reply:
I have not been a fan of employing cargo bininstall because it lacks supply chain security, doesnt use github attestations, compared to other tools in the space like homebrew, but to your point, it just also pulls from gh releases, which is no different to our existing approach and so doesn't actually have an advantage in regards to these things because it also doesn't do anything for supply chain security.

Let's run with this, but I'd like to circle back on it to making this more secure.

Reply:
I understand your concerns, and I will modify it to the previous way. bininstall has a [signing mechanism](https://github.com/cargo-bins/cargo-binstall/blob/main/SIGNING.md), but it seems not to be widely used.

Reply:
Let's leave it as you have it, as there's little advantage over it vs using bininstall, both suffer from the same problems. I'll merge this PR shortly.

Reply:
I'll have a little think and might change it in a follow up.